gcp-client/src/main/java/com/google/cloud/graphite/platforms/plugin/client/ContainerAnalysisClient.java

```diff
@@ -0,0 +1,189 @@
+   * @param publicKeyId The id of the public key for the asymmetric PIKIX key used to sign the
+   *     payload and produce the signature.
+   * @param payload The base64-encoded attestation verified by the provided signature.
+   * @return The attestation {@link Occurrence} that was created.
+   * @throws IOException An error occurred attempting to get the occurrence.
+   */
+  public Occurrence createAttestation(
+      final String projectId,
+      final String resourceUrl,
+      final String noteProjectId,
+      final String noteId,
+      final String signature,
+      final String publicKeyId,
+      final String payload)
+      throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(resourceUrl));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(noteProjectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(noteId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(signature));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(payload));
+    return containerAnalysis.createAttestation(
+        projectId, resourceUrl, noteProjectId, noteId, signature, publicKeyId, payload);
+  }
+
+  private Optional<Occurrence> getStatusOccurrence(String projectId, String resourceUrl) {
+    List<Occurrence> occurrences;
+    try {
+      occurrences =
+          containerAnalysis.listOccurrences(
+              projectId,
+              buildFilterString(
+                  ImmutableMap.of(
+                      "resourceUrl",
+                      resourceUrl,
+                      "noteProjectId",
+                      VULNERABILITY_NOTE_PROJECT_ID,
+                      "noteId",
+                      VULNERABILITY_NOTE_ID)));
+    } catch (IOException ioe) {
+      log.info(String.format("Error listing occurrences: %s. Retrying ...", ioe.getMessage()));
+      return Optional.empty();
+    }
+    for (Occurrence o : occurrences) {
+      if (o.getDiscovered() != null) {
+        return Optional.of(o);
+      }
+    }
+    log.info("Did not find occurrences.");
+    return Optional.empty();
+  }
+
+  private Optional<String> getVulnerabilityScanStatus(String projectId, String occurrenceId) {
+    Occurrence occurrence;
+    try {
+      occurrence = containerAnalysis.getOccurrence(projectId, occurrenceId);
+    } catch (IOException ioe) {
+      log.info(
+          String.format(
+              "Error retrieving vulnerability status occurrence projects/%s/occurrences/%s: %s",
+              projectId, occurrenceId, ioe.getMessage()));
+      return Optional.empty();
+    }
+    String status = occurrence.getDiscovered().getDiscovered().getAnalysisStatus();
+    if (FINISHED_STATUSES.contains(status)) {
+      return Optional.of(status);
+    }
+    log.info(String.format("Vulnerability scan is not finished. Current status is %s.", status));
```

Comment:
As this is expected to occur commonly, do we want to fill logs with it?

Reply:
Adjusted to log.fine, because this could be useful in certain contexts but it won't fill up the logs unless displaying fine logs is enabled.